annotations/raml2java.raml

```diff
@@ -0,0 +1,100 @@
+     description: |
+       Trivial mapping of RAML array types, is to map them to array types of the target language
+       however this behavior is not always desired. This annotation allows to customize
+       desired type of the container type when mapping array types. When used on type container
+       serves as default for all contained types.
+     default: list
+  javaName:
+     type: string
+     description: Allows to specify name of the generated java type or java property
+     allowedTargets: [TypeDeclaration]
+  implementsExisting:
+     type: string[]
+     description: |
+        Allows to specify additional existing interfaces which should be implemented by type.
+        Framework assumes that interface declarations already exists on class path
+     allowedTargets: [Library,TypeDeclaration,API,Overlay,Extension]
+  extendExisting:
+        type: string
+        description: |
+           Allows to specify existing superclass which should be extended by type.
+           Framework assumes that superclass declaration already exists on class path
+        allowedTargets: [Library,TypeDeclaration,API,Overlay,Extension]
+  mapsToExisting:
+        type: string
+        description: |
+           Allows to specify fully qualified name of existing class which should be mapped to this type
+        allowedTargets: [TypeDeclaration]
+  defaultNumberFormat:
+    description: This annotation allows to configure default java representation for number values
+    type: string
+    default: double
+    enum: [float, long, int, int16,int32, int64, double]
+    allowedTargets: [Library,TypeDeclaration,API,Overlay,Extension]
+  defaultIntegerFormat:
+    description: This annotation allows to configure default java representation for integer values
+    type: string
+    enum: [long, int, int16, int32, int64, int8]
+    default: int
+    allowedTargets: [Library,TypeDeclaration,API,Overlay,Extension]
+  cloneable:
+    type: nil
+    description: This marker annotation states that generated types should support cloning
+    allowedTargets: [Library,TypeDeclaration,API,Overlay,Extension]
+  serializable:
+    type: nil
+    description: This marker annotation states that generated types should be serializable
+    allowedTargets: [Library,TypeDeclaration,API,Overlay,Extension]
+  comparable:
+    #Not supported yet.
+    type: string
+    description: |
+       This annotation states that generated types should be comparable,
+       annotation argument is the name of property which should be used for comparison
+    allowedTargets: [Library,TypeDeclaration,API,Overlay,Extension]
+  equalsAndHashCode:
+    type: nil
+    description: |
+      This marker annotation states that generated types should have equals and hashCode methods
+      which supports comparison by value
+    allowedTargets: [Library,TypeDeclaration,API,Overlay,Extension]
+types:
+  JavaPackage: string
+  TypeName: string
+(commons.Authors):
```

Comment:
Not sure about that to be honest. Its kinda additional information that you might want to have for building a separate registry, but there might be better options than putting it into the fragment file. Again, if its not import for the consumer (in this case the application that reads those annotations), I would try to bring in the meta information in a different form.